tests/shared/drive-constraints.spec.js

```diff
@@ -0,0 +1,276 @@
+      });
+
+      m.chai.expect(result).to.be.true;
+    });
+
+    it('should return true if the drive size is equal to recommended size', function() {
+      const result = constraints.isDriveSizeRecommended({
+        device: '/dev/disk1',
+        name: 'USB Drive',
+        size: 2000000000,
+        protected: false
+      }, {
+        path: 'rpi.img',
+        size: 1000000000,
+        recommendedDriveSize: 2000000000
+      });
+
+      m.chai.expect(result).to.be.true;
+    });
+
+    it('should return false if the drive size is less than the recommended size', function() {
+      const result = constraints.isDriveSizeRecommended({
+        device: '/dev/disk1',
+        name: 'USB Drive',
+        size: 2000000000,
+        protected: false
+      }, {
+        path: 'rpi.img',
+        size: 1000000000,
+        recommendedDriveSize: 2000000001
+      });
+
+      m.chai.expect(result).to.be.false;
+    });
+
+    it('should throw if the image is undefined', function() {
+      const drive = {
+        device: '/dev/disk1',
+        name: 'USB Drive',
+        size: 1000000000,
+        protected: false
+      };
+
+      m.chai.expect(() => {
+        constraints.isDriveSizeRecommended(drive, undefined);
+      }).to.throw(TypeError);
+    });
+
+    it('should return true if the recommended drive size is undefined', function() {
+      const result = constraints.isDriveSizeRecommended({
+        device: '/dev/disk1',
+        name: 'USB Drive',
+        size: 2000000000,
+        protected: false
+      }, {
+        path: 'rpi.img',
+        size: 1000000000
+      });
+
+      m.chai.expect(result).to.be.true;
+    });
+
+  });
+
+  describe('.isDriveValid()', function() {
+
+    describe('given drive is large enough', function() {
+
+      beforeEach(function() {
+        this.drive = {
+          device: '/dev/disk2',
+          name: 'My Drive',
+          size: 4000000000
+        };
+        this.image = {
+          path: 'rpi.img',
+          size: 2000000000
+        };
+      });
+
+      it('should return true if drive is not locked', function() {
+        this.drive.protected = false;
+        m.chai.expect(constraints.isDriveValid(this.drive, this.image)).to.be.true;
+      });
+
+      it('should return false if drive is locked', function() {
+        this.drive.protected = true;
+        m.chai.expect(constraints.isDriveValid(this.drive, this.image)).to.be.false;
+      });
+    });
+    describe('given drive is not large enough', function() {
+
+      beforeEach(function() {
+        this.drive = {
+          device: '/dev/disk2',
+          name: 'My Drive',
+          size: 1000000000
+        };
+        this.image = {
+          path: 'rpi.img',
+          size: 2000000000
+        };
+      });
+
+      it('should return false', function() {
+        m.chai.expect(constraints.isDriveValid(this.drive, this.image)).to.be.false;
+      });
+    });
+
+    describe('given image is undefined', function() {
```

Comment:
Let's put blank lines between the `it` and `describe` blocks for consistency with the rest of the file:

```js
describe('...', function() {

    it('...', function() {

    });

});
```